content/docs/command-reference/repro.md

```diff
@@ -154,6 +155,9 @@ up-to-date and only execute the final stage.
   corresponding pipelines, including the target stages themselves. This option
   has no effect if `targets` are not provided.
 
+- `--pull` - try automatically [pulling](/doc/command-reference/pull) missing
+  cache for outputs restored from run-cache.
```

Comment:
Back on this. Per https://github.com/iterative/dvc/pull/4538#issuecomment-706765188:

> dvc repro --pull pulls regular files, hashes for which might've been restored from the existing run-cache, so kinda like regular dvc pull

Unfortunately I don't understand either one of the explanations. What's the relationship between run-cache and `repro --pull`? Maybe a step-by-step explanation like 1. Use `repro --pull`; 2. run-cache is checked before executing commands (default repro behavior I think); 3. Some output hashes are found? (but not the actual files? This is the confusing part); 4. Hashed are looked for in the cache but not found; 5. The files are looked for in remote storage. Something like that

Please @efiop ! Thanks in advance

Reply:
@jorgeorpinel Even if we leave the run-cache out, `repro --pull` would still try to `dvc pull` outputs that are missing, but the pipeline didn't change. E.g. when you forgot to `dvc pull` beforehand and you are trying to `dvc repro` otherwise up-to-date pipeline, so `dvc repro --pull` will just pull the outputs for such stages instead of trying to reproduce them.

Run-cache is then just a special source of lock files, and `repro --pull` works the same way as explained above.

Want to point out again that `--pull` is still a temporary solution that was needed to improve `pull --run-cache` that is also not complete in a product sense. So I would recommend not spending much time on this, as the product scenario is WIP and there is no reason to optimize the docs for it too much.

Reply:
OK it makes more sense now, thanks.

In this case I do feel like need to spend enough time understanding what's going on so that when the coming bulk of docs related to new features hit, I'm better prepared. So thanks again for baring with me!

Reply:
Last Q @efiop. Does this only check the default remote (if one is set)? Or all remotes?

Reply:
Actually, 2 more questions...

- Does it check only the local run-cache? Or also the remote run-cache for possible dep/out hashes?
- What happens if you do `repro --pull --no-run-cache`? Is the run-cache check skipped?

Thanks!

Reply:
> Does this only check the default remote (if one is set)? Or all remotes?

Only the default remote right now.

> Does it check only the local run-cache? Or also the remote run-cache for possible dep/out hashes?

Yes, only local run-cache.

> What happens if you do repro --pull --no-run-cache? Is the run-cache check skipped?

Correct. It will only pull if you have your lock file complete (so hashes are already there, just the outputs are missing from cache), but won't try to use run-cache.

Please feel free to ask any questions, I do understand that this incomplete feature is a bit confusing.

Reply:
That's all I can think of for now. Thanks @efiop! Updated in https://github.com/iterative/dvc.org/pull/1881/files#diff-6c1f3192f09e2722ba169e9fa219b3b5158bbafa470b382c2d6135db7aa1e20d.